website: remove usage of task network stanza (#8498)
hashicorp/nomad: website/pages/docs/job-specification/group.mdx

  all tasks in this group. If omitted, a default policy exists for each job
  type, which can be found in the [restart stanza documentation][restart].

- `service` <code>([Service][]: nil)</code> - Specifies integrations with

>   [Consul][] for service discovery. Nomad automatically registers when an allocation

Reviewer: ```suggestion
  [Consul][] for service discovery. Nomad automatically registers each service when an allocation
```

>   is started and de-registers it when the allocation dies.

Reviewer: ```suggestion
  is started and de-registers them when the allocation is destroyed.
```
